lib/chef/provisioning/convergence_strategy/precreate_chef_objects.rb

```diff
@@ -127,6 +127,10 @@ def create_ohai_files(action_handler, machine)
           convergence_options[:ohai_hints].each_pair do |hint, data|
             # The location of the ohai hint
             ohai_hint = "/etc/chef/ohai/hints/#{hint}.json"
+            # It's in a different path on windows
+            if machine.machine_spec.data['normal']['chef_provisioning']['reference']['is_windows']
```

Comment:
Sorry, just thought of this too - you should be able to do `if machine.machine_spec.reference['is_windows']` - will you try that and see if it works?